app/controllers/search_controller.rb

```diff
@@ -60,9 +85,81 @@ def reorder_filters(aggs, active_filters)
 
   def validate_q!
     return if params[:advanced]&.strip.present?
+    return if params[:geobox]&.strip.present?
+    return if params[:geodistance]&.strip.present?
     return if params[:q]&.strip.present?
 
     flash[:error] = 'A search term is required.'
     redirect_to root_url
   end
+
+  def validate_geodistance_presence!
+    return unless params[:geodistance]&.strip == 'true'
+
+    geodistance_params = [params[:geodistanceLatitude]&.strip, params[:geodistanceLongitude]&.strip,
+                          params[:geodistanceDistance]&.strip]
+    return if geodistance_params.all?(&:present?)
+
+    flash[:error] = 'All geospatial distance fields are required.'
+    redirect_to root_url
+  end
+
+  def validate_geobox_presence!
+    return unless params[:geobox]&.strip == 'true'
+
+    geobox_params = [params[:geoboxMinLatitude]&.strip, params[:geoboxMinLongitude]&.strip,
+                     params[:geoboxMaxLatitude]&.strip, params[:geoboxMaxLongitude]&.strip]
+    return if geobox_params.all?(&:present?)
+
+    flash[:error] = 'All bounding box fields are required.'
+    redirect_to root_url
+  end
+
+  def validate_geodistance_range!
+    return unless params[:geodistance]&.strip == 'true'
+
+    lat = params[:geodistanceLatitude]&.strip.to_f
+    long = params[:geodistanceLongitude]&.strip.to_f
+    return if lat.between?(-90.0, 90.0)
+    return if long.between?(-180.0, 180.0)
+
+    flash[:error] = 'Latitude must be between -90.0 and 90.0, and longitude must be -180.0 and 180.0.'
+    redirect_to root_url
+  end
+
+  def validate_geobox_range!
+    return unless params[:geobox]&.strip == 'true'
+
+    geobox_lat = [params[:geoboxMinLatitude]&.strip.to_f, params[:geoboxMaxLatitude]&.strip.to_f]
+    geobox_long = [params[:geoboxMinLongitude]&.strip.to_f, params[:geoboxMaxLongitude]&.strip.to_f]
+    return if geobox_lat.all? { |lat| lat.between?(-90.0, 90.0) }
+    return if geobox_long.all? { |long| long.between?(-180.0, 180.0) }
```

Comment:
Similar to the above, the return clauses here will only fail the validation if both of the latitude and longitude pairs are invalid. A single wrong value out of the four isn't sufficient to fail the query as a whole, because only one wrong value means that the query as a whole contains a valid pair of latitude or longitude values, and that valid pair will trigger the relevant return clause.